crates/async-dropper-derive/src/lib.rs

```diff
@@ -193,30 +201,37 @@ fn gen_impl(DeriveInput { ident, .. }: &DeriveInput) -> proc_macro2::TokenStream
                 // Spawn a task to do the drop
                 let task = ::async_std::task::spawn(async move {
                     let drop_fail_action = <#ident as ::async_dropper::AsyncDrop>::drop_fail_action(&original);
-                    match ::async_std::future::timeout(
+                    let task_res = match ::async_std::future::timeout(
                         <#ident as ::async_dropper::AsyncDrop>::drop_timeout(&original),
                         <#ident as ::async_dropper::AsyncDrop>::async_drop(&mut original),
                     ).await {
-                        Err(e) => {
+                        // Task timed out
+                        Err(_) | Ok(Err(AsyncDropError::Timeout)) => {
                             match drop_fail_action {
-                                ::async_dropper::DropFailAction::Continue => original,
-                                ::async_dropper::DropFailAction::Panic => {
-                                    panic!("async drop failed: {e}");
-                                }
+                                ::async_dropper::DropFailAction::Continue => Ok(()),
+                                ::async_dropper::DropFailAction::Panic => Err("async drop timed out".to_string()),
```

Comment:
```suggestion
                                ::async_dropper::DropFailAction::Continue => (original, Ok(())),
                                ::async_dropper::DropFailAction::Panic => (original, Err("async drop timed out".to_string())),
```

What do you think about getting rid of the temporary `let` binding for `task_res` and returning `(original, ...)` in every match arm?

I'm not firm on this, but wanted to bring it up as an option! Then at the end we can get rid of the `;` and additional `(original, task_res)`.

Really appreciate this fix & refactor by the way, effective and extremely easy to read and know what the cases are/what we're doing in these cases.

Reply:
Yeah, will update

Reply:
Checked this one more time. Don't think it is a good approach to put additional data in every `match` arm because of the repeated structure, but it is pretty subjective.

Reply:
I'm good with that, thanks for taking another look!